library/src/scala/tasty/Reflection.scala

```diff
@@ -1336,20 +1336,20 @@ trait Reflection extends reflect.Types { reflectSelf: CompilerInterface =>
   //   TYPES   //
   ///////////////
 
-  /** Returns the type (Type) of T */
-  def typeOf[T](using qtype: scala.quoted.Type[T], ctx: Context): Type =
-    qtype.asInstanceOf[scala.internal.quoted.Type[T]].typeTree.asInstanceOf[TypeTree].tpe
-
-
   // ----- Types ----------------------------------------------------
 
   given (using ctx: Context) as TypeTest[Type, Type] = reflectSelf.Type_TypeTest
   given TypeOps as Type.type = Type
 
   object Type:
 
-    def apply(clazz: Class[_])(using ctx: Context): Type =
-      reflectSelf.Type_apply(clazz)
+    /** Returns the type or kind (Type) of T */
+    def of[T <: AnyKind](using qtype: scala.quoted.Type[T], ctx: Context): Type =
+      qtype.asInstanceOf[scala.internal.quoted.Type[TypeTree]].typeTree.tpe
+
+    /** Returns the type or kind of the class of the type T */
+    def ofClass[T](using ct: scala.reflect.ClassTag[T]): Type =
```

Comment:
The existence of both `Type.of` and `Type.ofClass` can cause confusion. What about change `Type.of` to `Type.apply`. We can also change return type of `Type.of` to `Type[T]`:

```Scala
def apply[T <: AnyKind](using qtype: scala.quoted.Type[T]): Type[T] = qtype

// usage
Type[T]
```

I see in the test code we write `Type.of[T].seal` in many places. Without the change above, it's simpler.

Then we can rename `Type.ofClass` to `Type.of`.

Reply:
Can `Type.of` subsumes all usage of `Type.ofClass`?

Reply:
I did not notice the `Type.of[T].seal`. That is just bad code, they should be replaced with `'[T]` unless that are tests on the `seal` operation. I will check all those occurrences.

`Type.of` does subsume most cases of `Type.ofClass`. The only case it does not is if you get a `Class[_]` at runtime and want to recover the `Type` from it. The second is also slightly more performant.

The apply is not good because it leads to easy mistakes as `quoted.Type[T]` and `Reflect.Type[T]` dot not return the same thing. Knowing that you need to use `Type.of` helps users find the correct version and autocompletion may help to find the other variants that are more specific/performant.

Reply:
Actually `Type.of[T].seal` is equivalent to `'[T].asInstanceOf[Type[?]]` which is relly bad code.

Reply:
I only see 2 tests files with `Type.of[T].seal` where we explicitly test the `seal` operation.

Reply:
> The apply is not good because it leads to easy mistakes as `quoted.Type[T]` and `Reflect.Type[T]` dot not return the same thing. Knowing that you need to use `Type.of` helps users find the correct version and autocompletion may help to find the other variants that are more specific/performant.

I agree, it causes confusion.

> `Type.of` does subsume most cases of `Type.ofClass`. The only case it does not is if you get a `Class[_]` at runtime and want to recover the Type from it. The second is also slightly more performant.

If that is the case, can we simply remove `Type.ofClass` if there are no known use cases?
Or, at least rename it to a more awkward name, such as `Type.fromClassTag` to avoid conflicts with `Type.of`.

Reply:
In the library, it is used for `Liftable[Class[T]]`. The name should be similar to `Type.of` as it is doing a similar operation and those should be found together in auto-completion and documentation. The fact that it uses a `ClassTag` is an implementation detail that we should not leak in the name. This operation acts on the concept of a `Class[?]`, the use of the `ClassTag` is just to use it implicitly and avoid some syntactic overhead.

Reply:
The concern is exactly about usabbility in autocomplete: two methods pop up after `Type.of`, they overlap 90% use cases and have the same user-interface. This creates a [choice dilemma](https://en.wikipedia.org/wiki/Overchoice) for programmers.

I'd suggest either change the name or reduce the overlapping (e.g., `ofClass` can take an explicit argument).

Reply:
If it takes an explicit argument no one will use it as they would have to write much more code for something that they should be encouraged to do. `Type.ofClass[Foo]` would become `Type.ofClass(classOf[Foo])`. Additionally, that would introduce extra redundancy.

Reply:
This concept is in the middle between  `Type.of` and `classOf` and it makes sense to have a name that intuitively makes allusion to both those concepts.

Reply:
I think I have exhausted my arguments ;)
I'm happy to hear what others think about it.
I'm open to any agreed decision in order to move on.

/cc: @anatoliykmetyuk @bishabosha

Reply:
I don't mind the name but I am not sure what the benefit is except to conveniently have an unapplied polymorphic type

Reply:
That is unfortunately unavoidable for any `ClassTag` or `classOf`.